In [21]:
import pandas as pd

In [22]:
# Q1. Load your dataset into python using the pandas library and save data into a
# dataframe named dfi (where i is one of your datasets, for a total of 4).

csv_path1 = 'resources/electric_cars.csv'
df1 = pd.read_csv(csv_path1, encoding="utf-8")
df1.head()

Brand                          Model  AccelSec  TopSpeed_KmH  \
0       Tesla   Model 3 Long Range Dual Motor       4.6           233   
1  Volkswagen                       ID.3 Pure      10.0           160   
2    Polestar                               2       4.7           210   
3         BMW                            iX3        6.8           180   
4       Honda                              e        9.5           145   

   Range_Km  Efficiency_WhKm FastCharge_KmH RapidCharge PowerTrain  \
0       450              161            940         Yes        AWD   
1       270              167            250         Yes        RWD   
2       400              181            620         Yes        AWD   
3       360              206            560         Yes        RWD   
4       170              168            190         Yes        RWD   

     PlugType  BodyStyle Segment  Seats  PriceEuro  
0  Type 2 CCS      Sedan       D      5      55480  
1  Type 2 CCS  Hatchback       C      5      30000  
2  Type 2 CCS   Liftback       D      5      56440  
3  Type 2 CCS        SUV       D      5      68040  
4  Type 2 CCS  Hatchback       B      4      32997

In [23]:
csv_path2 = 'resources/kobe.csv'
df2 = pd.read_csv(csv_path2, encoding="utf-8")
df2.head()

vs  game quarter  time                                        description  \
0  ORL     1       1  9:47            Kobe Bryant makes 4-foot two point shot   
1  ORL     1       1  9:07                          Kobe Bryant misses jumper   
2  ORL     1       1  8:11                   Kobe Bryant misses 7-foot jumper   
3  ORL     1       1  7:41  Kobe Bryant makes 16-foot jumper (Derek Fisher...   
4  ORL     1       1  7:03                    Kobe Bryant makes driving layup   

  basket  
0      H  
1      M  
2      M  
3      H  
4      H

In [24]:
csv_path3 = 'resources/purchase_data.csv'
df3 = pd.read_csv(csv_path3, encoding="utf-8")
df3.head()

Purchase ID             SN  Age Gender  Item ID  \
0            0        Lisim78   20   Male      108   
1            1    Lisovynya38   40   Male      143   
2            2     Ithergue48   24   Male       92   
3            3  Chamassasya86   24   Male      100   
4            4      Iskosia90   23   Male      131   

                                   Item Name  Price  
0  Extraction, Quickblade Of Trembling Hands   3.53  
1                          Frenzied Scimitar   1.56  
2                               Final Critic   4.88  
3                                Blindscythe   3.27  
4                                       Fury   1.44

In [25]:
csv_path4 = 'resources/weather.csv'
df4 = pd.read_csv(csv_path4, encoding="utf-8")
df4.head()

Unnamed: 0 origin  year  month  day  hour   temp   dewp  humid  wind_dir  \
0           1    EWR  2013    1.0  1.0   0.0  37.04  21.92  53.97     230.0   
1           2    EWR  2013    1.0  1.0   1.0  37.04  21.92  53.97     230.0   
2           3    EWR  2013    1.0  1.0   2.0  37.94  21.92  52.09     230.0   
3           4    EWR  2013    1.0  1.0   3.0  37.94  23.00  54.51     230.0   
4           5    EWR  2013    1.0  1.0   4.0  37.94  24.08  57.04     240.0   

   wind_speed  wind_gust  precip  pressure  visib  
0    10.35702  11.918651     0.0    1013.9   10.0  
1    13.80936  15.891535     0.0    1013.0   10.0  
2    12.65858  14.567241     0.0    1012.6   10.0  
3    13.80936  15.891535     0.0    1012.7   10.0  
4    14.96014  17.215830     0.0    1012.8   10.0

In [26]:
# Q2. Preview your data by calling your dataframe’s name. How many columns and rows do you see?

df1.info()
df2.info()
df3.info()
df4.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 103 entries, 0 to 102
Data columns (total 14 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Brand            103 non-null    object 
 1   Model            103 non-null    object 
 2   AccelSec         103 non-null    float64
 3   TopSpeed_KmH     103 non-null    int64  
 4   Range_Km         103 non-null    int64  
 5   Efficiency_WhKm  103 non-null    int64  
 6   FastCharge_KmH   103 non-null    object 
 7   RapidCharge      103 non-null    object 
 8   PowerTrain       103 non-null    object 
 9   PlugType         103 non-null    object 
 10  BodyStyle        103 non-null    object 
 11  Segment          103 non-null    object 
 12  Seats            103 non-null    int64  
 13  PriceEuro        103 non-null    int64  
dtypes: float64(1), int64(5), object(8)
memory usage: 11.4+ KB
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 133 entries, 0 to 132
Data columns (total 6 co

In [27]:
# Q3. Examine the shape of your data using the .shape command and the data types of
# your columns using .dtypes.

df1.shape
df2.shape
df3.shape
df4.shape

(8719, 15)

In [28]:
df1.dtypes


Brand               object
Model               object
AccelSec           float64
TopSpeed_KmH         int64
Range_Km             int64
Efficiency_WhKm      int64
FastCharge_KmH      object
RapidCharge         object
PowerTrain          object
PlugType            object
BodyStyle           object
Segment             object
Seats                int64
PriceEuro            int64
dtype: object

In [29]:
df2.dtypes

vs             object
game            int64
quarter        object
time           object
description    object
basket         object
dtype: object

In [30]:
df3.dtypes

Purchase ID      int64
SN              object
Age              int64
Gender          object
Item ID          int64
Item Name       object
Price          float64
dtype: object

In [31]:
df4.dtypes

Unnamed: 0      int64
origin         object
year            int64
month         float64
day           float64
hour          float64
temp          float64
dewp          float64
humid         float64
wind_dir      float64
wind_speed    float64
wind_gust     float64
precip        float64
pressure      float64
visib         float64
dtype: object

In [32]:
# Q4. Use .describe() on your data. What do you notice about your data? What does this
# command return?

# Ans: It returns the statistical summary analysis of each column in the dataframe.

df1.describe()


AccelSec  TopSpeed_KmH    Range_Km  Efficiency_WhKm       Seats  \
count  103.000000    103.000000  103.000000       103.000000  103.000000   
mean     7.396117    179.194175  338.786408       189.165049    4.883495   
std      3.017430     43.573030  126.014444        29.566839    0.795834   
min      2.100000    123.000000   95.000000       104.000000    2.000000   
25%      5.100000    150.000000  250.000000       168.000000    5.000000   
50%      7.300000    160.000000  340.000000       180.000000    5.000000   
75%      9.000000    200.000000  400.000000       203.000000    5.000000   
max     22.400000    410.000000  970.000000       273.000000    7.000000   

           PriceEuro  
count     103.000000  
mean    55811.563107  
std     34134.665280  
min     20129.000000  
25%     34429.500000  
50%     45000.000000  
75%     65000.000000  
max    215000.000000

In [33]:
df2.describe()

game
count  133.000000
mean     2.902256
std      1.450521
min      1.000000
25%      1.000000
50%      3.000000
75%      4.000000
max      5.000000

In [34]:
df3.describe()

Purchase ID         Age     Item ID       Price
count   780.000000  780.000000  780.000000  780.000000
mean    389.500000   22.714103   91.755128    3.050987
std     225.310896    6.659444   52.697702    1.169549
min       0.000000    7.000000    0.000000    1.000000
25%     194.750000   20.000000   47.750000    1.980000
50%     389.500000   22.000000   92.000000    3.150000
75%     584.250000   25.000000  138.000000    4.080000
max     779.000000   45.000000  183.000000    4.990000

In [35]:
df4.describe()

Unnamed: 0    year        month          day         hour  \
count  8719.000000  8719.0  8718.000000  8718.000000  8718.000000   
mean   4360.000000  2013.0     6.506538    15.679628    11.515715   
std    2517.102832     0.0     3.439961     8.768250     6.916846   
min       1.000000  2013.0     1.000000     1.000000     0.000000   
25%    2180.500000  2013.0     4.000000     8.000000     6.000000   
50%    4360.000000  2013.0     7.000000    16.000000    12.000000   
75%    6539.500000  2013.0     9.000000    23.000000    18.000000   
max    8719.000000  2013.0    12.000000    31.000000    23.000000   

              temp         dewp        humid     wind_dir   wind_speed  \
count  8718.000000  8718.000000  8718.000000  8486.000000  8718.000000   
mean     55.484942    41.794184    62.863333   193.194674     9.335732   
std      18.345817    19.619776    19.572699   110.541333    12.328261   
min      10.940000    -9.040000    13.950000     0.000000     0.000000   
25%      39.920000    26.060000    47.602500   110.000000     5.753900   
50%      55.940000    42.800000    62.330000   220.000000     9.206240   
75%      71.060000    57.920000    78.790000   280.000000    12.658580   
max     100.040000    75.920000   100.000000   360.000000  1048.360580   

         wind_gust       precip     pressure        visib  
count  8718.000000  8719.000000  7780.000000  8719.000000  
mean     10.743374     0.003035  1017.826465     9.224522  
std      14.187117     0.022686     7.396722     2.094044  
min       0.000000     0.000000   983.900000     0.120000  
25%       6.621473     0.000000  1012.800000    10.000000  
50%      10.594357     0.000000  1017.500000    10.000000  
75%      14.567241     0.000000  1022.900000    10.000000  
max    1206.432388     1.180000  1041.900000    10.000000

In [36]:
# Q5. Use the .head() and .tail() command - what does this do?

# Ans: .head() fetches the first 5 rows in the dataframe.
# .tail() fetches the last 5 rows in the dataframe.

df1.head()

Brand                          Model  AccelSec  TopSpeed_KmH  \
0       Tesla   Model 3 Long Range Dual Motor       4.6           233   
1  Volkswagen                       ID.3 Pure      10.0           160   
2    Polestar                               2       4.7           210   
3         BMW                            iX3        6.8           180   
4       Honda                              e        9.5           145   

   Range_Km  Efficiency_WhKm FastCharge_KmH RapidCharge PowerTrain  \
0       450              161            940         Yes        AWD   
1       270              167            250         Yes        RWD   
2       400              181            620         Yes        AWD   
3       360              206            560         Yes        RWD   
4       170              168            190         Yes        RWD   

     PlugType  BodyStyle Segment  Seats  PriceEuro  
0  Type 2 CCS      Sedan       D      5      55480  
1  Type 2 CCS  Hatchback       C      5      30000  
2  Type 2 CCS   Liftback       D      5      56440  
3  Type 2 CCS        SUV       D      5      68040  
4  Type 2 CCS  Hatchback       B      4      32997

In [37]:
df1.tail()

Brand                            Model  AccelSec  TopSpeed_KmH  \
98   Nissan                       Ariya 63kWh       7.5           160   
99     Audi     e-tron S Sportback 55 quattro       4.5           210   
100  Nissan               Ariya e-4ORCE 63kWh       5.9           200   
101  Nissan   Ariya e-4ORCE 87kWh Performance       5.1           200   
102   Byton                 M-Byte 95 kWh 2WD       7.5           190   

     Range_Km  Efficiency_WhKm FastCharge_KmH RapidCharge PowerTrain  \
98        330              191            440         Yes        FWD   
99        335              258            540         Yes        AWD   
100       325              194            440         Yes        AWD   
101       375              232            450         Yes        AWD   
102       400              238            480         Yes        AWD   

       PlugType  BodyStyle Segment  Seats  PriceEuro  
98   Type 2 CCS  Hatchback       C      5      45000  
99   Type 2 CCS        SUV       E      5      96050  
100  Type 2 CCS  Hatchback       C      5      50000  
101  Type 2 CCS  Hatchback       C      5      65000  
102  Type 2 CCS        SUV       E      5      62000

In [38]:
df2.head()

vs  game quarter  time                                        description  \
0  ORL     1       1  9:47            Kobe Bryant makes 4-foot two point shot   
1  ORL     1       1  9:07                          Kobe Bryant misses jumper   
2  ORL     1       1  8:11                   Kobe Bryant misses 7-foot jumper   
3  ORL     1       1  7:41  Kobe Bryant makes 16-foot jumper (Derek Fisher...   
4  ORL     1       1  7:03                    Kobe Bryant makes driving layup   

  basket  
0      H  
1      M  
2      M  
3      H  
4      H

In [39]:
df2.tail()

vs  game quarter     time                         description basket
128  ORL     3       4     3:33         Bryant Layup Shot: Missed        M
129  ORL     3       4     2:02           Bryant 3pt Shot: Missed        M
130  ORL     3       4  00:23.9           Bryant 3pt Shot: Missed        M
131  ORL     3       4  00:06.9           Bryant 3pt Shot: Missed        M
132  ORL     3       4  00:00.5  Bryant Layup Shot: Made (31 PTS)        H

In [40]:
df3.head()

Purchase ID             SN  Age Gender  Item ID  \
0            0        Lisim78   20   Male      108   
1            1    Lisovynya38   40   Male      143   
2            2     Ithergue48   24   Male       92   
3            3  Chamassasya86   24   Male      100   
4            4      Iskosia90   23   Male      131   

                                   Item Name  Price  
0  Extraction, Quickblade Of Trembling Hands   3.53  
1                          Frenzied Scimitar   1.56  
2                               Final Critic   4.88  
3                                Blindscythe   3.27  
4                                       Fury   1.44

In [41]:
df3.tail()

Purchase ID          SN  Age  Gender  Item ID  \
775          775  Aethedru70   21  Female       60   
776          776      Iral74   21    Male      164   
777          777  Yathecal72   20    Male       67   
778          778     Sisur91    7    Male       92   
779          779   Ennrian78   24    Male       50   

                                 Item Name  Price  
775                                   Wolf   3.54  
776                       Exiled Doomblade   1.63  
777  Celeste, Incarnation of the Corrupted   3.46  
778                           Final Critic   4.19  
779                                   Dawn   4.60

In [42]:
df4.head()

Unnamed: 0 origin  year  month  day  hour   temp   dewp  humid  wind_dir  \
0           1    EWR  2013    1.0  1.0   0.0  37.04  21.92  53.97     230.0   
1           2    EWR  2013    1.0  1.0   1.0  37.04  21.92  53.97     230.0   
2           3    EWR  2013    1.0  1.0   2.0  37.94  21.92  52.09     230.0   
3           4    EWR  2013    1.0  1.0   3.0  37.94  23.00  54.51     230.0   
4           5    EWR  2013    1.0  1.0   4.0  37.94  24.08  57.04     240.0   

   wind_speed  wind_gust  precip  pressure  visib  
0    10.35702  11.918651     0.0    1013.9   10.0  
1    13.80936  15.891535     0.0    1013.0   10.0  
2    12.65858  14.567241     0.0    1012.6   10.0  
3    13.80936  15.891535     0.0    1012.7   10.0  
4    14.96014  17.215830     0.0    1012.8   10.0

In [43]:
df4.tail()

Unnamed: 0 origin  year  month   day  hour   temp   dewp  humid  \
8714        8715    JFK  2013    9.0   2.0  20.0  75.20  73.40  94.14   
8715        8716    JFK  2013   10.0  23.0  10.0  48.92  39.02  68.51   
8716        8717    JFK  2013   10.0  23.0  11.0  48.92  39.02  68.51   
8717        8718    JFK  2013   12.0  17.0   5.0  26.96  10.94  50.34   
8718        8719    LGA  2013    8.0  22.0  22.0  75.92  66.92  73.68   

      wind_dir  wind_speed  wind_gust  precip  pressure  visib  
8714     200.0     4.60312   5.297178     0.0       NaN    4.0  
8715      60.0     4.60312   5.297178     0.0    1008.1   10.0  
8716      40.0     4.60312   5.297178     0.0    1008.5   10.0  
8717      40.0     4.60312   5.297178     0.0    1023.9   10.0  
8718     210.0     8.05546   9.270062     0.0    1011.9   10.0

**## Extra Credit (3 pts):**

In [44]:
# list(df3 columns) or
list(df3.columns)

['Purchase ID', 'SN', 'Age', 'Gender', 'Item ID', 'Item Name', 'Price']

In [45]:
# 1. Choose one of your datasets and remove the header information. (Can delete
# the row in excel, etc..)

# Ans: In this DataFrame, the original header of the input CSV has been ignored, and the first row of the input data has been set as a header.
# One metod is: header = False
# and another is: skiprows = 1

csv_path = 'resources/weather.csv'
df3 = pd.read_csv(csv_path4, encoding="utf-8", skiprows = 1)

In [46]:
# 2. Import the data into your environment using pandas. Display the .head() of your
# data showing no header information.

df3.head()

1  EWR  2013  1.1  1.2    0  37.04  21.92  53.97    230  10.35702  \
0  2  EWR  2013  1.0  1.0  1.0  37.04  21.92  53.97  230.0  13.80936   
1  3  EWR  2013  1.0  1.0  2.0  37.94  21.92  52.09  230.0  12.65858   
2  4  EWR  2013  1.0  1.0  3.0  37.94  23.00  54.51  230.0  13.80936   
3  5  EWR  2013  1.0  1.0  4.0  37.94  24.08  57.04  240.0  14.96014   
4  6  EWR  2013  1.0  1.0  6.0  39.02  26.06  59.37  270.0  10.35702   

   11.9186514756  0.1  1013.9    10  
0      15.891535  0.0  1013.0  10.0  
1      14.567241  0.0  1012.6  10.0  
2      15.891535  0.0  1012.7  10.0  
3      17.215830  0.0  1012.8  10.0  
4      11.918651  0.0  1012.0  10.0

In [47]:
# 3. Using pandas, update the dataset to include the header information. Display the
# updated data using .head().

# adding column name to the respective columns
df3.columns = [
  'index',
 'origin',
 'year',
 'month',
 'day',
 'hour',
 'temp',
 'dewp',
 'humid',
 'wind_dir',
 'wind_speed',
 'wind_gust',
 'precip',
 'pressure',
 'visib']

In [48]:
df3.head()

index origin  year  month  day  hour   temp   dewp  humid  wind_dir  \
0      2    EWR  2013    1.0  1.0   1.0  37.04  21.92  53.97     230.0   
1      3    EWR  2013    1.0  1.0   2.0  37.94  21.92  52.09     230.0   
2      4    EWR  2013    1.0  1.0   3.0  37.94  23.00  54.51     230.0   
3      5    EWR  2013    1.0  1.0   4.0  37.94  24.08  57.04     240.0   
4      6    EWR  2013    1.0  1.0   6.0  39.02  26.06  59.37     270.0   

   wind_speed  wind_gust  precip  pressure  visib  
0    13.80936  15.891535     0.0    1013.0   10.0  
1    12.65858  14.567241     0.0    1012.6   10.0  
2    13.80936  15.891535     0.0    1012.7   10.0  
3    14.96014  17.215830     0.0    1012.8   10.0  
4    10.35702  11.918651     0.0    1012.0   10.0

**## Extra Credit (3 pts)**
## Data Wrangling:

In [62]:
# 3. Using pandas, update the dataset to include the header information. Display the
# updated data using .head().
csv_path = 'resources/atu_dirty.csv'
dirty_df = pd.read_csv(csv_path, encoding="utf-8")
dirty_df.head()

Unnamed: 0          caseid  V1  V2         V3  V4   V5  V6   V7
0           1  20160101160045  62   2  No answer   1  690   0  465
1           2  20160101160066  69   1  No answer   2  600   0  560
2           3  20160101160069  24   2  No answer   1  940   0   20
3           4  20160101160083  31   2  Part time   1  635   0  120
4           5  20160101160084  59   2  Part time   1  500   0  177

In [63]:
dirty_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10493 entries, 0 to 10492
Data columns (total 9 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   Unnamed: 0  10493 non-null  int64 
 1   caseid      10493 non-null  int64 
 2   V1          10493 non-null  int64 
 3   V2          10493 non-null  int64 
 4   V3          10493 non-null  object
 5   V4          10493 non-null  int64 
 6   V5          10493 non-null  int64 
 7   V6          10493 non-null  int64 
 8   V7          10493 non-null  int64 
dtypes: int64(8), object(1)
memory usage: 737.9+ KB


**## atu_dirty Documentation:**
American Time Use Survey Data Sample - Dirty
Description
A dataset containing a subset of variables from the American Time Use Survey. This dataset is "dirty", meaning it has elements which require formatting before use.

Usage
data(atu_dirty)
Format
A data frame with 10,493 observations of 8 variables

Details
caseid. unique identifier of individual survey participant

V1. the age of the respondent

V2. the gender of the respondent (1: Male, 2: Female)

V3. the employment status of the respondent

V4. does the respondent have a physical difficulty (1: Person did not report having a physical difficulty, 2: Person surveyed reported the have a physical difficulty)

V5. the length of time the person sleeps, in minutes

V6. How long the respondent spent on homework assignments, in minutes

V7. the number of minutes the respondent spent socializing

In [64]:
# Dropping the "unnamed: 0" column

dirty_df = dirty_df.drop("Unnamed: 0",axis=1)

dirty_df.head()

caseid  V1  V2         V3  V4   V5  V6   V7
0  20160101160045  62   2  No answer   1  690   0  465
1  20160101160066  69   1  No answer   2  600   0  560
2  20160101160069  24   2  No answer   1  940   0   20
3  20160101160083  31   2  Part time   1  635   0  120
4  20160101160084  59   2  Part time   1  500   0  177

In [65]:
# Assigning column names
dirty_df.columns = ["case_id", "age", "gender", "employment_status", "physical_difficulty", "sleep_time(min)", "hw_time_spent(min)", "socializing_time(min)"]
dirty_df.head()

case_id  age  gender employment_status  physical_difficulty  \
0  20160101160045   62       2         No answer                    1   
1  20160101160066   69       1         No answer                    2   
2  20160101160069   24       2         No answer                    1   
3  20160101160083   31       2         Part time                    1   
4  20160101160084   59       2         Part time                    1   

   sleep_time(min)  hw_time_spent(min)  socializing_time(min)  
0              690                   0                    465  
1              600                   0                    560  
2              940                   0                     20  
3              635                   0                    120  
4              500                   0                    177

In [68]:
# To convert column values to lower case

dirty_df['employment_status'] = dirty_df['employment_status'].str.lower()

In [69]:
dirty_df.head()

case_id  age  gender employment_status  physical_difficulty  \
0  20160101160045   62       2         no answer                    1   
1  20160101160066   69       1         no answer                    2   
2  20160101160069   24       2         no answer                    1   
3  20160101160083   31       2         part time                    1   
4  20160101160084   59       2         part time                    1   

   sleep_time(min)  hw_time_spent(min)  socializing_time(min)  
0              690                   0                    465  
1              600                   0                    560  
2              940                   0                     20  
3              635                   0                    120  
4              500                   0                    177

In [70]:
dirty_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10493 entries, 0 to 10492
Data columns (total 8 columns):
 #   Column                 Non-Null Count  Dtype 
---  ------                 --------------  ----- 
 0   case_id                10493 non-null  int64 
 1   age                    10493 non-null  int64 
 2   gender                 10493 non-null  int64 
 3   employment_status      10493 non-null  object
 4   physical_difficulty    10493 non-null  int64 
 5   sleep_time(min)        10493 non-null  int64 
 6   hw_time_spent(min)     10493 non-null  int64 
 7   socializing_time(min)  10493 non-null  int64 
dtypes: int64(7), object(1)
memory usage: 655.9+ KB


In [71]:
# Checking null values. Using isnull() function 
dirty_df.isnull()


case_id    age  gender  employment_status  physical_difficulty  \
0        False  False   False              False                False   
1        False  False   False              False                False   
2        False  False   False              False                False   
3        False  False   False              False                False   
4        False  False   False              False                False   
...        ...    ...     ...                ...                  ...   
10488    False  False   False              False                False   
10489    False  False   False              False                False   
10490    False  False   False              False                False   
10491    False  False   False              False                False   
10492    False  False   False              False                False   

       sleep_time(min)  hw_time_spent(min)  socializing_time(min)  
0                False               False                  False  
1                False               False                  False  
2                False               False                  False  
3                False               False                  False  
4                False               False                  False  
...                ...                 ...                    ...  
10488            False               False                  False  
10489            False               False                  False  
10490            False               False                  False  
10491            False               False                  False  
10492            False               False                  False  

[10493 rows x 8 columns]

In [72]:
# using isnull() function 
dirty_df.describe()

case_id           age        gender  physical_difficulty  \
count  1.049300e+04  10493.000000  10493.000000         10493.000000   
mean   2.016063e+13     49.418565      1.554941             1.124369   
std    3.486162e+08     18.112827      0.496996             0.330017   
min    2.016010e+13     15.000000      1.000000             1.000000   
25%    2.016030e+13     35.000000      1.000000             1.000000   
50%    2.016061e+13     49.000000      2.000000             1.000000   
75%    2.016091e+13     64.000000      2.000000             1.000000   
max    2.016121e+13     85.000000      2.000000             2.000000   

       sleep_time(min)  hw_time_spent(min)  socializing_time(min)  
count     10493.000000        10493.000000           10493.000000  
mean        534.614791            7.046507             298.430477  
std         137.164066           45.883855             213.629890  
min           0.000000            0.000000               0.000000  
25%         450.000000            0.000000             130.000000  
50%         525.000000            0.000000             260.000000  
75%         600.000000            0.000000             432.000000  
max        1428.000000          980.000000            1392.000000